talk/morelanguage/morestl.tex

```diff
@@ -147,6 +147,35 @@
   \end{exerciseWithShortcut}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[23]{std::expected \cpprefLink{https://en.cppreference.com/w/cpp/utility/expected}}
+  \begin{block}{Manages either of 2 values (expected or not)}
+    \begin{itemize}
+    \item templated by the 2 value types
+    \item Useful for the return value of a function that may fail
+      \begin{itemize}
+      \item and would then return another type (error type)
+      \end{itemize}
+    \end{itemize}
+  \end{block}
+  \begin{exampleblock}{}
+    \small
+    \begin{cppcode*}{}
+      enum class Error {...};
+      std::expected<int, Error> parse(std::string_view in) {
+        if (is_valid(in)) return convert_to_int(in);
+        return std::expected(Error::...);
```

Comment:
```suggestion
        return std::unexpected(Error::...);
```
For the un-happy path, I think the intention is to use `std::unexpected`, although I have no experience with `std::expected` yet :)

This is also shown in the example on cppreference: https://en.cppreference.com/w/cpp/utility/expected